pilot/pkg/model/push_context.go

```diff
@@ -1627,32 +1630,31 @@ func (ps *PushContext) initDefaultExportMaps() {
 // with the proxy and derive listeners/routes/clusters based on the sidecar
 // scope.
 func (ps *PushContext) initSidecarScopes(env *Environment) {
-	sidecarConfigs := env.List(gvk.Sidecar, NamespaceAll)
+	rawSidecarConfigs := env.List(gvk.Sidecar, NamespaceAll)
 
-	sortConfigByCreationTime(sidecarConfigs)
+	sortConfigByCreationTime(rawSidecarConfigs)
 
-	sidecarConfigWithSelector := make([]config.Config, 0)
-	sidecarConfigWithoutSelector := make([]config.Config, 0)
-	for _, sidecarConfig := range sidecarConfigs {
+	sidecarConfigs := make([]config.Config, 0, len(rawSidecarConfigs))
+	for _, sidecarConfig := range rawSidecarConfigs {
 		sidecar := sidecarConfig.Spec.(*networking.Sidecar)
+		// sidecars with selector take preference
 		if sidecar.WorkloadSelector != nil {
-			sidecarConfigWithSelector = append(sidecarConfigWithSelector, sidecarConfig)
-		} else {
-			sidecarConfigWithoutSelector = append(sidecarConfigWithoutSelector, sidecarConfig)
+			sidecarConfigs = append(sidecarConfigs, sidecarConfig)
+		}
+	}
+	for _, sidecarConfig := range rawSidecarConfigs {
```

Comment:
Is this really more efficient with two loops vs one loop previously?

Reply:
My benchmark results say "yes", my test code is [here](https://github.com/wulianglongrd/playground/blob/main/demos/merge_test.go#L21).

Note: it's not always efficient, it's related to the overhead inside the for loop code block, it's very cheap here (just check if `sidecar.WorkloadSelector` is nil).